include/nbl/asset/utils/IMeshPacker.h

```diff
@@ -275,15 +273,13 @@ class IMeshPacker : public IMeshPackerBase
 
         //triangle reordering
         {
-            //this is needed for mesh buffers with no index buffer (triangle strips and triagnle fans)
-            //TODO: fix
-            //bool wasTmpIdxBufferSet = false;
-            //if (meshBuffer->getIndexBufferBinding().buffer == nullptr)
-            //{
-            //    //temporary use generated index buffer
-            //    wasTmpIdxBufferSet = true;
-            //    meshBuffer->setIndexBufferBinding(idxBufferParams.idxBuffer);
-            //}
+            const bool wasTmpIdxBufferSet = idxBufferParams.idxBuffer.buffer.get() != nullptr ? true : false;
+            if (wasTmpIdxBufferSet)
+            {
+                auto tmpBuffer = std::move(meshBuffer->getIndexBufferBinding());
+                meshBuffer->setIndexBufferBinding(std::move(idxBufferParams.idxBuffer));
+                idxBufferParams.idxBuffer = std::move(tmpBuffer);
+            }
```

Comment:
instead of screwing around with the index buffer pinned to the `meshBuffer` maybe sidestep the
`auto triangleIndices = IMeshManipulator::getTriangleIndices(meshBuffer, ix++);`
with your own implementation that uses the new index buffer ?

Reply:
ok